fldigi: new, 4.1.20

WhiredPlanck opened the pull request:
<!-- For description on topic creation and maintenance, please refer to [this Wiki article](https://wiki.aosc.io/developer/packaging/topic-based-maintenance-guideline/). -->

Topic Description
-----------------

Add `fldigi`, a Digital Modem Program for Amateur Radio.

Package(s) Affected
-------------------

- `aprx`
- `fldigi`
- `hamlib`
- `libax25`
- `qsstv`

Security Update?
----------------

<!-- If this topic is part of a security update, please uncomment "Yes,"
     and mark with the `security` label, as well as reference issue number below for priority processing. -->

<!-- Yes - Issue Number: ISSUENUMBER -->
No

<!-- Please uncomment the "Build Order" section if applicable, this is commonly needed in package updates/introduction that affects more than one package. -->


Build Order
-----------

- `hamlib` -> `fldigi` or `qsstv`

Architectural Progress
----------------------

<!-- Please remove any architecture to which this topic does not apply. -->

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`
<!-- If this package involves a `+32` counterpart, please uncomment the line below. -->    
<!-- - [ ] 32-bit Optional Environment `optenv32` -->

<!-- If all package(s) affected by this topic is `noarch`, please use the stub below. -->
<!-- - [ ] Architecture-independent `noarch` -->

Secondary Architectural Progress
--------------------------------

<!-- Please remove any architecture to which this topic does not apply. -->

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`

----

After the pull request is merged, all package(s) affected must be rebuilt against the `stable` Git tree and environment (only `stable` repository should be enabled in `sources.list`). This section marks the progress above.

Please, make sure the list of architectures below matches the ones above.

Post-Merge Architectural Progress
---------------------------------

<!-- Please remove any architecture to which this topic does not apply. -->

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`
<!-- If this package involves a `+32` counterpart, please uncomment the line below. -->
<!-- - [ ] 32-bit Optional Environment `optenv32` -->

<!-- If all package(s) affected by this topic is `noarch`, please use the stub below. -->
<!-- - [ ] Architecture-independent `noarch` -->

Post-Merge Secondary Architectural Progress
-------------------------------------------

<!-- Please remove any architecture to which this topic does not apply. -->

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`

<!-- TODO: CI to auto-fill architectural progress. -->

WhiredPlanck committed:
fldigi: new, 4.1.20

WhiredPlanck committed:
fldigi: adjust PKGDEP

WhiredPlanck committed:
aprx: new, 2.9.0

WhiredPlanck committed:
hamlib: update to 4.2

WhiredPlanck committed:
libax25: new, 1.1.1

WhiredPlanck committed:
qsstv: update to 9.5.8

WhiredPlanck committed:
qsstv: add patch file

WhiredPlanck commented:
fldigi, amd64, LGTM
![图片](https://user-images.githubusercontent.com/47623588/131215587-0de9013f-6a12-4ebb-83db-59196dc2327a.png)

qsstv, amd64, LGTM
![图片](https://user-images.githubusercontent.com/47623588/131215635-ce4e347c-c2ae-4456-8503-c588eb6416a6.png)

WhiredPlanck committed:
aprx: remove autobuild/build

    - Utilise ABTYPE instead.
    - Disable shadow build.

WhiredPlanck committed:
fldigi: use versioning variable in source link

WhiredPlanck committed:
libax25: add the comment for the action in autobuild/beyond

WhiredPlanck committed:
libax25: start PKGDES with a upper-case letter

WhiredPlanck committed:
qsstv: modify the autobuild/patch

    - Add the comment for the action.
    - Reference the file to be patched with absolute path.

WhiredPlanck committed:
libax25: replace source link

jyhi commented:
Jeff will look after this.

WhiredPlanck commented:
> Jeff will look after this.

Thanks ~

WhiredPlanck committed:
libax25: remove unnecessary blank spaces

MingcongBai commented on extra-radio/aprx/autobuild/prepare:
```diff
@@ -0,0 +1,2 @@
+abinfo 'Append -g to CFLAGS to make this have debug symbols and -fcommon -O2 to compile with gcc 10'
+export CFLAGS="-Wall -g -fcommon -O2"
```
You should be using Autobuild3-defined flags, and with that, you'd only need to append `-fcommon`. Do the following:

```
export CFLAGS="${CFLAGS} -fcommon"
```

And adjust your `abinfo` accordingly.

Also, almost all of your files lack `newline` markers - Visual Studio Code?

WhiredPlanck replied:
Sorry to make you laugh. I used `$CFLAGS` before but I didn't keep it for no reason ...
In addition, I used KWrite to make all modification. It's a little wired, because I notice almost every local file has `newline`.

MingcongBai commented on extra-radio/fldigi/autobuild/defines:
```diff
@@ -0,0 +1,7 @@
+PKGDEP="fltk hamlib libpng libsndfile libsamplerate portaudio pulseaudio"
+BUILDDEP="asciidoc"
+
+AUTOTOOLS_AFTER="--prefix=/usr"
```
This is already defined in `AUTOTOOLS_DEF`. See `/etc/autobuild/ab3_defcfg.sh`.

MingcongBai commented on extra-radio/hamlib/spec:
```diff
@@ -1,5 +1,4 @@
-VER=3.3
+VER=4.2
```
Rebuilds necessary?

MingcongBai commented on extra-radio/libax25/autobuild/beyond:
```diff
@@ -0,0 +1,2 @@
+abinfo 'Fedora: Removing those headers conflict with glibc-headers...'
+rm -f "$PKGDIR"/usr/include/{netax25/ax25.h,netrom/netrom.h,netrose/rose.h}
```
Add verbosity here - `rm -v`. Don't use `-f`, it's irresponsible.

MingcongBai commented on extra-radio/libax25/autobuild/defines:
```diff
@@ -0,0 +1,8 @@
+PKGDES="Library for AX.25, ROSE, 6PACK, KISS and NETROM protocols"
+PKGDEP="gcc zlib"
+
+AUTOTOOLS_AFTER="--prefix=/usr \
```
All of these are already defined in `AUTOTOOLS_DEF`.

WhiredPlanck replied:
I removed these in other packages, but I forgot this package ...... Will fix soon.

MingcongBai commented on extra-radio/libax25/spec:
```diff
@@ -0,0 +1,4 @@
+VER=1.1.1
+SRCS="https://github.com/ve7fet/linuxax25/archive/libax15-$VER.tar.gz"
+CHKSUMS="sha256::6e0eb9f4160df57a09131e5c123adf40ecb972aacdc44706e6f9079b2a5eef76"
+SUBDIR=linuxax25-libax15-$VER/libax25
```
Quote your paths.

MingcongBai commented on extra-radio/qsstv/autobuild/beyond:
```diff
@@ -1 +1 @@
-install -Dm644 "$SRCDIR"/qsstv/icons/qsstv.png "$PKGDIR"/usr/share/pixmaps/qsstv.png
+install -Dm644 "$SRCDIR"/icons/qsstv.png "$PKGDIR"/usr/share/pixmaps/qsstv.png
```
Add verbosity, and `abinfo` here.

MingcongBai replied:
I'd also suggest wrapping long lines with `\`, this makes your code easier to read. Personally I go no longer than 80 characters.

WhiredPlanck replied:
Get it. Here I just simply corrected the path. Will fix soon.

MingcongBai commented on extra-radio/qsstv/autobuild/patch:
```diff
@@ -0,0 +1,2 @@
+abinfo 'Patching the qsstv.pro: rename qwt-qt5 to qwt after LIBS variale...'
+sed -i 's/LIBS += \-lqwt\-qt5/LIBS += \-lqwt/g' "$SRCDIR"/qsstv.pro
```
I'd suggest that you make your `sed` commands clearer, use full `-e` and `-i` expressions, and wrap long lines. Here's how I'd write this:

```
sed -e 's/LIBS += \-lqwt\-qt5/LIBS += \-lqwt/g' \
    -i "$SRCDIR"/qsstv.pro
```

WhiredPlanck replied:
I referenced this from other packages' autobuild/patch, most of them use one-line command. But obviously this suggestion looks more reasonable.

MingcongBai commented on extra-radio/qsstv/autobuild/patch:
```diff
@@ -0,0 +1,2 @@
+abinfo 'Patching the qsstv.pro: rename qwt-qt5 to qwt after LIBS variale...'
```
Good morning, your personal grammar nazi here:

- "The" before "qsstv.pro" is not necessary.
- Variable *

MingcongBai commented on extra-radio/aprx/autobuild/defines:
```diff
@@ -0,0 +1,6 @@
+PKGNAME=aprx
+PKGSEC=hamradio
+PKGDES="A highly configurable APRS I-gate/Digipeater Daemon"
```
Try not to advertise the upstream's claims, they might say it's "highly configurable," but not everyone will agree.

MingcongBai commented on extra-radio/fldigi/autobuild/defines:
```diff
@@ -0,0 +1,7 @@
+PKGNAME=fldigi
+PKGSEC=hamradio
+PKGDES="Digital Modem Program for Amateur Radio"
```
Why are all words capitalised? They shouldn't be.

MingcongBai commented on extra-radio/fldigi/spec:
```diff
@@ -0,0 +1,4 @@
+VER=4.1.20
+SRCS="tbl::https://sourceforge.net/projects/fldigi/files/fldigi/fldigi-$VER.tar.gz/download"
```
Drop `/download` at the end.

WhiredPlanck committed:
aprx: mark newline in spec

WhiredPlanck committed:
aprx: use defined CLAGS instead in autobuild/prepare

WhiredPlanck committed:
aprx: don't advertise upstream claims in PKGDES

WhiredPlanck committed:
fldigi: adjust source link

WhiredPlanck committed:
fldigi: modify autobuild/defines

    - Only uppercase the first letter of the sentence in PKGDES.
    - Remove unnecessary build type variables.

WhiredPlanck committed:
libax25: show the action's verbosity in autobuild/beyond

WhiredPlanck committed:
hamlib: mark PKGBREAK

WhiredPlanck committed:
gpredict: bump REL

WhiredPlanck committed:
libax25: remove unnecessary build type variables

WhiredPlanck committed:
libax25: quote SUBDIR path

WhiredPlanck committed:
qsstv: add abinfo and verbosity to the action in autobuild/beyond

WhiredPlanck committed:
qsstv: make the action in autobuild/patch clearer
